File: Front/src/app/services/roteiro.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Roteiro {
  id?: number;
  titulo: string;
  destino: string;
  dataInicio: string;
  dataFim: string;
  custoEstimado: number;
}

@Injectable({
  providedIn: 'root'
})
export class RoteiroService {
  private apiUrl = '/api/roteiros';

  constructor(private http: HttpClient) { }

  getRoteiros(): Observable<Roteiro[]> {
    return this.http.get<Roteiro[]>(this.apiUrl);
  }

  criarRoteiro(roteiro: Roteiro): Observable<Roteiro> {
    return this.http.post<Roteiro>(this.apiUrl, roteiro);
  }

  atualizarRoteiro(roteiro: Roteiro): Observable<Roteiro> {
    return this.http.put<Roteiro>(`${this.apiUrl}/${roteiro.id}`, roteiro);
  }

  deletarRoteiro(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: Back/roteiro-service/src/main/java/com/roteiro/roteiro_service/model/Roteiro.java
package com.roteiro.roteiro_service.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;

@Entity
@Data
public class Roteiro {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String titulo;
    private String destino;
    private String dataInicio;
    private String dataFim;
    private Double custoEstimado;


}